src/main/java/team18/team18_be/apply/service/ApplyService.java

```diff
@@ -1,25 +1,108 @@
+import team18.team18_be.resume.repository.ResumeRepository;
 import team18.team18_be.userInformation.dto.request.ApplicationFormRequest;
+import team18.team18_be.userInformation.entity.Company;
+import team18.team18_be.userInformation.entity.ForeignerInformation;
+import team18.team18_be.userInformation.repository.CompanyRepository;
+import team18.team18_be.userInformation.repository.ForeignerInformationRepository;
 
 @Service
 public class ApplyService {
 
   private final ApplicationFormRepository applicationFormRepository;
+  private final ApplyRepository applyRepository;
+  private final RecruitmentRepository recruitmentRepository;
+  private final ResumeRepository resumeRepository;
+  private final CompanyRepository companyRepository;
+  private final ForeignerInformationRepository foreignerInformationRepository;
 
-  public ApplyService(ApplicationFormRepository applicationFormRepository) {
+
+  public ApplyService(ApplicationFormRepository applicationFormRepository,
+      ApplyRepository applyRepository, RecruitmentRepository recruitmentRepository,
+      ResumeRepository resumeRepository, CompanyRepository companyRepository,
+      ForeignerInformationRepository foreignerInformationRepository) {
     this.applicationFormRepository = applicationFormRepository;
+    this.applyRepository = applyRepository;
+    this.recruitmentRepository = recruitmentRepository;
+    this.resumeRepository = resumeRepository;
+    this.companyRepository = companyRepository;
+    this.foreignerInformationRepository = foreignerInformationRepository;
+  }
+
+  public Long createApplicationForm(ApplicationFormRequest applicationFormRequest,
+      Long recruitmentId, User user) {
+    String status = "지원 중";
+    Recruitment recruitment = recruitmentRepository.findById(recruitmentId)
+        .orElseThrow(() -> new NoSuchElementException("해당되는 구인글이 없습니다."));
+    Apply apply = new Apply(status, user, recruitment);
+    Apply savedApply = applyRepository.save(apply);
+
+    return savedApply.getId();
+  }
+
+
+  public List<ApplierPerRecruitmentResponse> searchApplicant(Long recruitmentId, User user) {
+    Recruitment recruitment = recruitmentRepository.findById(recruitmentId)
+        .orElseThrow(() -> new NoSuchElementException("해당되는 구인글이 없습니다."));
+    List<Apply> applyList = applyRepository.findByRecruitment(recruitment);
+    List<ApplierPerRecruitmentResponse> applyResponseList = new ArrayList<>();
+    for (Apply apply : applyList) {
+      User applicantUser = apply.getUser(); //구인글에 지원한 지원자
+      Resume resume = resumeRepository.findByUser(applicantUser); //그 지원자의 이력서 가져오기
+      ApplierPerRecruitmentResponse applierPerRecruitmentResponse = new ApplierPerRecruitmentResponse(
+          applicantUser.getId(), applicantUser.getName(), resume.getResumeId(), apply.getId(),
+          "베트남", resume.getKorean());
+      applyResponseList.add(applierPerRecruitmentResponse);
+    }
+    return applyResponseList;
+  }
```

Comment:
이런식으로 고차함수를 사용하여 코드를 깔끔하게 작성할 수 있겠네요!
```suggestion
  public List<ApplierPerRecruitmentResponse> searchApplicant(Long recruitmentId, User user) {
    Recruitment recruitment = getEntityOrThrow(recruitmentRepository.findById(recruitmentId), "해당되는 구인글이 없습니다.");

    return applyRepository.findByRecruitment(recruitment).stream()
        .map(this::createApplierPerRecruitmentResponse)  // 각 Apply 객체를 변환
        .collect(Collectors.toList());
  }

  private ApplierPerRecruitmentResponse createApplierPerRecruitmentResponse(Apply apply) {
    User applicantUser = apply.getUser();
    Resume resume = getEntityOrThrow(Optional.ofNullable(resumeRepository.findByUser(applicantUser)), "해당 지원자의 이력서를 찾을 수 없습니다.");

    return new ApplierPerRecruitmentResponse(
        applicantUser.getId(), applicantUser.getName(), resume.getResumeId(), apply.getId(),
        "베트남", resume.getKorean()
    );
  }
```

Reply:
훨씬 간결해지는 것 같습니다! 반영하겠습니다!!